compiler/executable/match_/planner/plan.rs

```diff
@@ -941,42 +949,47 @@ impl PartialCostPlan {
                     graph.elements[&pattern_id].is_valid(pattern_id, &all_available_vars, graph)
                 }
             })
-            .flat_map(move |&extension| {
-                let join_var = self.determine_joinability(graph, extension);
-
-                if join_var.is_none() {
-                    vec![(extension, join_var)].into_iter()
-                } else {
-                    vec![(extension, None), (extension, join_var)].into_iter()
-                }
+            .flat_map(move |&extension| match self.determine_joinability(graph, extension) {
+                None => vec![(extension, None)].into_iter(),
+                Some(var) => vec![(extension, None), (extension, Some(var))].into_iter(),
```

Comment:
One more step: `.flat_map` accepts `IntoIterator`
```suggestion
                None => vec![(extension, None)],
                Some(var) => vec![(extension, None), (extension, Some(var))],
```